homeassistant/components/sensor/isy994.py

```diff
@@ -1,95 +1,310 @@
-    for (folder_name, states) in (
-            ('HA.locations', [STATE_HOME, STATE_NOT_HOME]),
-            ('HA.sensors', [STATE_OPEN, STATE_CLOSED]),
-            ('HA.states', [STATE_ON, STATE_OFF])):
-        try:
-            folder = ISY.programs['My Programs'][folder_name]
-        except KeyError:
-            # folder does not exist
-            pass
-        else:
-            for _, _, node_id in folder.children:
-                node = folder[node_id].leaf
-                devs.append(ISYSensorDevice(node, states))
+    devices = []
+
+    for node in SENSOR_NODES:
+        if (len(node.uom) == 0 or node.uom[0] not in BINARY_UOM) and \
+                STATE_OFF not in node.uom and STATE_ON not in node.uom:
+            _LOGGER.debug('LOADING %s', node.name)
+            devices.append(ISYSensorDevice(node))
 
-    add_devices(devs)
+    add_devices(devices)
 
 
-class WeatherPseudoNode(object):
-    """This class allows weather variable to act as regular nodes."""
+class ISYSensorDevice(ISYDevice):
+    """Representation of an ISY994 sensor device."""
 
-    # pylint: disable=too-few-public-methods
-    def __init__(self, device_id, name, status, units=None):
-        """Initialize the sensor."""
-        self._id = device_id
-        self.name = name
-        self.status = status
-        self.units = units
+    def __init__(self, node) -> None:
+        """Initialize the ISY994 sensor device."""
+        ISYDevice.__init__(self, node)
 
+    @property
+    def raw_unit_of_measurement(self) -> str:
+        """Get the raw unit of measurement for the ISY994 sensor device."""
+        if len(self._node.uom) == 1:
+            if self._node.uom[0] in UOM_FRIENDLY_NAME:
+                friendly_name = UOM_FRIENDLY_NAME.get(self._node.uom[0])
+                if friendly_name == TEMP_CELSIUS or \
+                        friendly_name == TEMP_FAHRENHEIT:
+                    friendly_name = self.hass.config.units.temperature_unit
+                return friendly_name
+            else:
+                return self._node.uom[0]
+        else:
+            return None
 
-class ISYSensorDevice(ISYDeviceABC):
-    """Representation of an ISY sensor."""
+    @property
+    def state(self) -> str:
+        """Get the state of the ISY994 sensor device."""
+        if len(self._node.uom) == 1:
+            if self._node.uom[0] in UOM_TO_STATES:
+                states = UOM_TO_STATES.get(self._node.uom[0])
+                if self.value in states:
+                    return states.get(self.value)
+            elif self._node.prec and self._node.prec != [0]:
+                str_val = str(self.value)
+                int_prec = int(self._node.prec)
+                decimal_part = str_val[-int_prec:]
+                whole_part = str_val[:len(str_val) - int_prec]
+                val = float('{}.{}'.format(whole_part, decimal_part))
+                raw_units = self.raw_unit_of_measurement
+                if raw_units in (
```

Comment:
I'm doing it because the line length will be too long - and we do need to know the raw units.

`unit_of_measurement` is converted to what the UI should display
`val` is the raw value which could need to be converted for UI

So I pull the `raw_unit_of_measurement` so I can convert from that to the expected.